teps/0048-task-results-without-results.md

```diff
@@ -283,6 +284,35 @@ spec:
       value: $(tasks.propagate-image-name.results.image)
 ```
 
+#### Providing default values for optional `Results`
+
+Today, a `Task` can declare a `Result` but not produce the `Result`
+and it's execution will still be successful. If a subsequent `Task`
+attempts to use the `Result`, then it would fail (causing the `Pipeline` to fail).
+As such, a `Result` is optional when unused but required when used - this
+can be confusing to users when failures happen later in the `Pipeline`.
+Supporting default values for the optional `Results` would help prevent 
+unexpected failures, and improve our `Task` interface`.
+
+For example, the following `Task` will produce an optional
+result when the none of the `Steps` in the `Task` produces the
+results:
+
+```yaml
+...
+spec:
+  ...
+  results:
+    - name: revision
+      description: Revision which was checkout. (branch, tag, sha, ref, etc...)
+      default: "main"
```

Comment:
We have not signed off on the proposal yet but I think its ok to include this new field here to demonstrate the use case.